core/src/lib/bsys.cc

```diff
@@ -233,7 +233,7 @@ char* bstrinlinecpy(char* dest, const char* src)
  */
 char* bstrncpy(char* dest, const char* src, int maxlen)
 {
-  strncpy(dest, src, maxlen - 1);
+  memmove(dest, src, maxlen - 1);
```

Comment:
this may overflow while reading `src` if `dest` is larger and `maxlen > sizeof(src)`.
I think we need to use `min(maxlen - 1, strlen(src) + 1)` so we copy only the src string including its null-terminator or up to maxlen.

Reply:
Now another approach is implemented.